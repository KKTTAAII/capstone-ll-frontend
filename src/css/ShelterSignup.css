.ShelterSignup-container {
    margin-top: 75px;
}

.ShelterSignup-header {
    font-size: 2em;
    margin: 20px;
    text-align: center;
}

.ShelterSignup-form {
    display: flex;
    flex-direction: column;
    width: 1000px;
    align-items: center;
    justify-content: center;
    margin: 0 auto;
}

.ShelterSignup-label {
    font-weight: 400;
    text-transform: uppercase;
    margin-bottom: 5px;
}

.row {
    width: 800px;
    display: flex;
    justify-content: space-between;
}

#bio {
    margin-top: 20px;
}

.ShelterSignup-input {
    width: 200px;
    background-color: transparent;
    border: none;
    color: black;
    outline: none;
    border-bottom: 1px solid #000000;
    font-size: 1em;
    font-weight: 300;
    padding-bottom: 10px;
    margin-top: 10px;
    margin-bottom: 10px;
}

.ShelterSignup-button {
    border-radius: 10px;
    border-style: none;
    background-color: white;
    height: 40px;
    width: 70px;
    margin: 15px;
    box-shadow: rgba(0, 0, 0, 0.25) 0px 54px 55px, rgba(0, 0, 0, 0.12) 0px -12px 30px, rgba(0, 0, 0, 0.12) 0px 4px 6px, rgba(0, 0, 0, 0.17) 0px 12px 13px, rgba(0, 0, 0, 0.09) 0px -3px 5px;
}

.ShelterSignup-button:hover {
    box-shadow: rgba(0, 0, 0, 0.07) 0px 1px 2px, rgba(0, 0, 0, 0.07) 0px 2px 4px, rgba(0, 0, 0, 0.07) 0px 4px 8px, rgba(0, 0, 0, 0.07) 0px 8px 16px, rgba(0, 0, 0, 0.07) 0px 16px 32px, rgba(0, 0, 0, 0.07) 0px 32px 64px;
}

.ShelterSignup-input-long {
    width: 500px;
    background-color: transparent;
    border: none;
    color: black;
    outline: none;
    border-bottom: 1px solid #000000;
    font-size: 1em;
    font-weight: 300;
    padding-bottom: 10px;
    margin-top: 10px;
    margin-bottom: 10px;
}

.ShelterSignup-warning {
    font-size: x-small;
    color: red;
    margin: 10px;
    font-weight: 400;
}

.ShelterSignup-select {
    border-radius: 5px;
}

.ShelterSignup-alreadySignup {
    text-align: center;
}

.ShelterSignup-signupLink {
    text-decoration: none;
    color: rgb(226, 129, 148);
}