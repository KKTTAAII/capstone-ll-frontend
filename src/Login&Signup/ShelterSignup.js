import React, { useState } from "react";
import { Link, useHistory } from "react-router-dom";
import swal from "sweetalert";
import {
  checkAllRequiredField,
  createInput,
  WARNING,
  createStateOptions,
  USSTATES,
} from "../common/helpers";
import DEFAULT_PIC from "../assets/shelter.jpg";
import "../css/ShelterSignup.css";
import { Row, Col, FormGroup } from "reactstrap";

const INITIAL_STATE = {
  username: "",
  password: "",
  name: "",
  address: "",
  city: "",
  state: "",
  postcode: "",
  phoneNumber: "",
  email: "",
  logo: "",
  description: "",
};

const ShelterSignUp = ({ signUp }) => {
  const [formData, setFormData] = useState(INITIAL_STATE);
  const [isTouched, setIsTouched] = useState(false);
  const [isInvalid, setIsInvalid] = useState(true);
  const history = useHistory();

  const handleChange = e => {
    const { name, value } = e.target;
    setFormData(formData => ({
      ...formData,
      [name]: value,
    }));
    setIsTouched(true);
    e.target.value === "" ? setIsInvalid(true) : setIsInvalid(false);
  };

  const handleSubmit = async e => {
    e.preventDefault();
    setFormData(INITIAL_STATE);
    const {
      username,
      password,
      name,
      city,
      state,
      phoneNumber,
      email,
    } = formData;
    const isAllRequiredFieldFilled = checkAllRequiredField([
      username,
      password,
      name,
      city,
      state,
      phoneNumber,
      email,
    ]);

    formData.logo = formData.logo === "" ? DEFAULT_PIC : formData.logo;

    if (!isInvalid && isAllRequiredFieldFilled) {
      let response = await signUp("Shelter", formData);
      //if there is a response, there is an error
      if (response) {
        swal({ text: response[0], icon: "warning" });
      } else {
        history.push("/shelters");
      }
    } else {
      swal({
        text: "Oops, please fill out all required fields",
        icon: "warning",
      });
      console.log("Oops, please fill out all required fields");
    }
  };

  return (
    <div className="ShelterSignup-container">
      <div className="ShelterSignup-header">Shelter Sign up</div>

      <form onSubmit={handleSubmit} className="ShelterSignup-form">
        <Row>
          <Col md={6}>
            <FormGroup>
              {createInput(
                "username",
                "text",
                formData.username,
                handleChange,
                "Username",
                true,
                "ShelterSignup-label",
                "ShelterSignup-input"
              )}
            </FormGroup>
          </Col>
          <Col md={6}>
            <FormGroup>
              {createInput(
                "password",
                "password",
                formData.password,
                handleChange,
                "Password",
                true,
                "ShelterSignup-label",
                "ShelterSignup-input"
              )}
            </FormGroup>
          </Col>
        </Row>

        <Row>
          <Col md={4}>
            <FormGroup>
              {createInput(
                "name",
                "text",
                formData.name,
                handleChange,
                "Shelter Name",
                true,
                "ShelterSignup-label",
                "ShelterSignup-input"
              )}
            </FormGroup>
          </Col>
          <Col md={4}>
            <FormGroup>
              {createInput(
                "email",
                "email",
                formData.email,
                handleChange,
                "Email",
                true,
                "ShelterSignup-label",
                "ShelterSignup-input"
              )}
            </FormGroup>
          </Col>
          <Col md={4}>
            <FormGroup>
              {createInput(
                "phoneNumber",
                "text",
                formData.phoneNumber,
                handleChange,
                "Phone",
                true,
                "ShelterSignup-label",
                "ShelterSignup-input"
              )}
            </FormGroup>
          </Col>
        </Row>

        <Row>
          <Col md={3}>
            <FormGroup>
              {createInput(
                "address",
                "text",
                formData.address,
                handleChange,
                "Address",
                false,
                "ShelterSignup-label",
                "ShelterSignup-input"
              )}
            </FormGroup>
          </Col>
          <Col md={3}>
            <FormGroup>
              {createInput(
                "city",
                "text",
                formData.city,
                handleChange,
                "City",
                true,
                "ShelterSignup-label",
                "ShelterSignup-input"
              )}
            </FormGroup>
          </Col>
          <Col md={1}>
            <FormGroup>
              <label htmlFor="state" className="ShelterSignup-label">
                State:
              </label>
              <select
                id="state"
                name="state"
                onChange={handleChange}
                value={formData.state}
                className="ShelterSignup-select"
              >
                {createStateOptions(USSTATES)}
              </select>
            </FormGroup>
          </Col>
          <Col md={2}>
            <FormGroup>
              {createInput(
                "postcode",
                "text",
                formData.postcode,
                handleChange,
                "Post Code",
                false,
                "ShelterSignup-label",
                "ShelterSignup-input"
              )}
            </FormGroup>
          </Col>
        </Row>

        {createInput(
          "logo",
          "text",
          formData.logo,
          handleChange,
          "Shelter's Logo URL",
          false,
          "ShelterSignup-label",
          "ShelterSignup-input-long"
        )}

        <label className="ShelterSignup-label" id="bio">
          Shelter's Bio:
        </label>
        <textarea
          id="description"
          name="description"
          value={formData.description}
          onChange={handleChange}
          rows="5"
          cols="33"
        ></textarea>
        {isInvalid && isTouched && (
          <small className="ShelterSignup-warning">{WARNING}</small>
        )}
        <button className="ShelterSignup-button">Sign up</button>
      </form>

      <div className="ShelterSignup-alreadySignup">
        Already signed up?{" "}
        <Link to={`/shelters/login`} className="ShelterSignup-signupLink">
          Log in
        </Link>
      </div>
    </div>
  );
};

export default ShelterSignUp;
