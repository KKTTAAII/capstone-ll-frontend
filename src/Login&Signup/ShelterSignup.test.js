import React from "react";
import { render } from "@testing-library/react";
import ShelterSignup from "./ShelterSignup";
import { MemoryRouter } from "react-router-dom";


it("renders without crashing", () => {
  render(
    <MemoryRouter>
      <ShelterSignup />
    </MemoryRouter>
  );
});

it("matches snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <ShelterSignup />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
