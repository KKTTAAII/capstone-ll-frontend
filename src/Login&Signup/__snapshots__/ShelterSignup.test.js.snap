// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ShelterSignup-container"
  >
    <div
      class="ShelterSignup-header"
    >
      Shelter Sign up
    </div>
    <form
      class="ShelterSignup-form"
    >
      <label
        class="ShelterSignup-label"
      >
        Username*:
      </label>
      <input
        class="ShelterSignup-input"
        id="username"
        name="username"
        placeholder="Username"
        type="text"
        value=""
      />
      <label
        class="ShelterSignup-label"
      >
        Password*:
      </label>
      <input
        class="ShelterSignup-input"
        id="password"
        name="password"
        placeholder="Password"
        type="password"
        value=""
      />
      <label
        class="ShelterSignup-label"
      >
        Shelter Name*:
      </label>
      <input
        class="ShelterSignup-input"
        id="name"
        name="name"
        placeholder="Shelter Name"
        type="text"
        value=""
      />
      <label
        class="ShelterSignup-label"
      >
        Email*:
      </label>
      <input
        class="ShelterSignup-input"
        id="email"
        name="email"
        placeholder="Email"
        type="email"
        value=""
      />
      <label
        class="ShelterSignup-label"
      >
        Phone*:
      </label>
      <input
        class="ShelterSignup-input"
        id="phoneNumber"
        name="phoneNumber"
        placeholder="Phone"
        type="text"
        value=""
      />
      <label
        class="ShelterSignup-label"
      >
        Address:
      </label>
      <input
        class="ShelterSignup-input"
        id="address"
        name="address"
        placeholder="Address"
        type="text"
        value=""
      />
      <label
        class="ShelterSignup-label"
      >
        City*:
      </label>
      <input
        class="ShelterSignup-input"
        id="city"
        name="city"
        placeholder="City"
        type="text"
        value=""
      />
      <label
        class="ShelterSignup-label"
        for="state"
      >
        State:
      </label>
      <select
        class="ShelterSignup-select"
        id="state"
        name="state"
      >
        <option
          value="AL"
        >
          AL
        </option>
        <option
          value="AK"
        >
          AK
        </option>
        <option
          value="AS"
        >
          AS
        </option>
        <option
          value="AZ"
        >
          AZ
        </option>
        <option
          value="AR"
        >
          AR
        </option>
        <option
          value="CA"
        >
          CA
        </option>
        <option
          value="CO"
        >
          CO
        </option>
        <option
          value="CT"
        >
          CT
        </option>
        <option
          value="DE"
        >
          DE
        </option>
        <option
          value="DC"
        >
          DC
        </option>
        <option
          value="FM"
        >
          FM
        </option>
        <option
          value="FL"
        >
          FL
        </option>
        <option
          value="GA"
        >
          GA
        </option>
        <option
          value="GU"
        >
          GU
        </option>
        <option
          value="HI"
        >
          HI
        </option>
        <option
          value="ID"
        >
          ID
        </option>
        <option
          value="IL"
        >
          IL
        </option>
        <option
          value="IN"
        >
          IN
        </option>
        <option
          value="IA"
        >
          IA
        </option>
        <option
          value="KS"
        >
          KS
        </option>
        <option
          value="KY"
        >
          KY
        </option>
        <option
          value="LA"
        >
          LA
        </option>
        <option
          value="ME"
        >
          ME
        </option>
        <option
          value="MH"
        >
          MH
        </option>
        <option
          value="MD"
        >
          MD
        </option>
        <option
          value="MA"
        >
          MA
        </option>
        <option
          value="MI"
        >
          MI
        </option>
        <option
          value="MN"
        >
          MN
        </option>
        <option
          value="MS"
        >
          MS
        </option>
        <option
          value="MO"
        >
          MO
        </option>
        <option
          value="MT"
        >
          MT
        </option>
        <option
          value="NE"
        >
          NE
        </option>
        <option
          value="NV"
        >
          NV
        </option>
        <option
          value="NH"
        >
          NH
        </option>
        <option
          value="NJ"
        >
          NJ
        </option>
        <option
          value="NM"
        >
          NM
        </option>
        <option
          value="NY"
        >
          NY
        </option>
        <option
          value="NC"
        >
          NC
        </option>
        <option
          value="ND"
        >
          ND
        </option>
        <option
          value="MP"
        >
          MP
        </option>
        <option
          value="OH"
        >
          OH
        </option>
        <option
          value="OK"
        >
          OK
        </option>
        <option
          value="OR"
        >
          OR
        </option>
        <option
          value="PW"
        >
          PW
        </option>
        <option
          value="PA"
        >
          PA
        </option>
        <option
          value="PR"
        >
          PR
        </option>
        <option
          value="RI"
        >
          RI
        </option>
        <option
          value="SC"
        >
          SC
        </option>
        <option
          value="SD"
        >
          SD
        </option>
        <option
          value="TN"
        >
          TN
        </option>
        <option
          value="TX"
        >
          TX
        </option>
        <option
          value="UT"
        >
          UT
        </option>
        <option
          value="VT"
        >
          VT
        </option>
        <option
          value="VI"
        >
          VI
        </option>
        <option
          value="VA"
        >
          VA
        </option>
        <option
          value="WA"
        >
          WA
        </option>
        <option
          value="WV"
        >
          WV
        </option>
        <option
          value="WI"
        >
          WI
        </option>
        <option
          value="WY"
        >
          WY
        </option>
      </select>
      <label
        class="ShelterSignup-label"
      >
        Post Code:
      </label>
      <input
        class="ShelterSignup-input"
        id="postcode"
        name="postcode"
        placeholder="Post Code"
        type="text"
        value=""
      />
      <label
        class="ShelterSignup-label"
      >
        Shelter's Logo URL:
      </label>
      <input
        class="ShelterSignup-input-long"
        id="logo"
        name="logo"
        placeholder="Shelter's Logo URL"
        type="text"
        value=""
      />
      <label
        class="ShelterSignup-label"
        id="bio"
      >
        Shelter's Bio:
      </label>
      <textarea
        cols="33"
        id="description"
        name="description"
        rows="5"
      />
      <button
        class="ShelterSignup-button"
      >
        Sign up
      </button>
    </form>
    <div
      class="ShelterSignup-alreadySignup"
    >
      Already signed up? 
      <a
        class="ShelterSignup-signupLink"
        href="/shelters/login"
      >
        Log in
      </a>
    </div>
  </div>
</DocumentFragment>
`;
